### ISSUE TITLE
Support importing `.geojson` files

### ISSUE THREAD

**Issue by nicolo-ribaudo** [denoland/deno#20176]
`import ... with { type: "json" }` allows loading any file that has a JSON MIME type (step 9 of HTML's [fetch a single module script](https://html.spec.whatwg.org/#fetch-a-single-module-script)). A JSON MIME type is `application/json`, `text/json`, or any that ends in `+json` (https://mimesniff.spec.whatwg.org/#json-mime-type).

GeoJSON files have `application/geo+json` MIME type (https://www.iana.org/assignments/media-types/application/geo+json), and as such HTML supports them.

It would be great if Deno could align with browsers on this, by allowing imports of `.geojson` files. Note that I didn't test what happens in Deno when loading a remote file with `application/geo+json` MIME type, but only when loading local file with the `.geojson` extension.

**Comment by bartlomieju** [comment]
This should be quite easy to fix, but will require changes to `deno_graph` crate. The problem is that `.geojson` extension is not considered a `MediaType::Json` but `MediaType::Unknown`. Once this is fixed in `deno_graph` then everything should work. If anyone is interested in tackling this I can provide more pointers.

**Comment by 43656D** [comment]
> This should be quite easy to fix, but will require changes to `deno_graph` crate. The problem is that `.geojson` extension is not considered a `MediaType::Json` but `MediaType::Unknown`. Once this is fixed in `deno_graph` then everything should work. If anyone is interested in tackling this I can provide more pointers.

I am interested in this issue, I need more details

**Comment by bartlomieju** [comment]
You'll need to open a PR to `deno_media_type` crate first, that updates discovery of `MediaType` enum here:
https://github.com/denoland/deno_media_type/blob/55310ec8c1f85c54bfe11f57f1a1c96156fcb668/src/lib.rs#L229

It should return `Json` if the extension is `.geojson`. Tests at the bottom of the file should be updated as well.

**Comment by 43656D** [comment]
I've open a [pull request](https://github.com/denoland/deno_media_type/pull/4). 
Do I need to add a different enum for the formatter? Or is it enough to just return `Json` with the `.geojson` for import support?

https://github.com/denoland/deno_media_type/blob/main/src/lib.rs#L288

**Comment by dsherret** [comment]
> do you feel we should add specialized GeoJson enum variant?

I don't think so. It will increase the chance of bugs. How popular is geojson? Should we even add this?

**Comment by bartlomieju** [comment]
> > do you feel we should add specialized GeoJson enum variant?
> 
> I don't think so. It will increase the chance of bugs. How popular is geojson? Should we even add this? Adding support for `+json` seems good, but couldn't someone rename a `.geojson` file to `.json`?

It appears it's used by GIS applications as well as some map software like Leaflet

**Comment by dsherret** [comment]
I think we should just add support for `+json` in the content type and not add another file extension for importing JSON. This seems very niche and I believe people can just use .json extensions for GeoJSON anyway.

**Comment by bartlomieju** [comment]
Given that the change to support it is 3 lines of code I would err we actually support it. Seems like an unnecessary papercut that is almost free to get rid of.

**Comment by dsherret** [comment]
It's not three lines of code. It will be handling it in many places.

**Comment by bartlomieju** [comment]
Sorry, I was imprecise. Let's not add `GeoJson` variant, but let's recognize `.geojson` extension in `MediaType` (https://github.com/denoland/deno_media_type/pull/4).

**Comment by dsherret** [comment]
I was saying it's more than 3 lines of code to support the new extension. It's not just updating deno_media_type. I don't think we should support an additional extension for importing JSON.

**Comment by bartlomieju** [comment]
Fair enough, I don't have a vested interest in this, I'm fine with just `+json` content-type.

**Comment by 43656D** [comment]
So should I use a regex (or any expression) to return `Json` for each mime type with `+json` (`application/vnd.github.raw+json`, `application/geo+json`) extension for content-type.

**Comment by lucacasonato** [comment]
I think supporting .geojson when imported remotely when it has the content type set automatically by a server to `application/geo+json`, but not when imported locally is weird and inconsistent. 

I do think we should support `.geojson` locally. For that matter, I think we should not use direct extension matching for `.geojson` specifically (this would open the door to adding support for every other file extension encoded in JSON manually), but instead we use the IANA MIME type registry to first map extensions to MIME types, and then perform the `application/json` / `application/*+json` check on this mime type.

We already have [this mimedb registry in std](https://deno.land/std@0.200.0/media_types/vendor/mime-db.v1.52.0.ts), it should be trivial to pull that into `deno_media_types`.

**Comment by lucacasonato** [comment]
> It's not just updating deno_media_type

@dsherret What else needs to be updated?

**Comment by dsherret** [comment]
> I think supporting .geojson when imported remotely when it has the content type set automatically by a server to application/geo+json, but not when imported locally is weird and inconsistent.

Why? The content type ends with `+json`.

**Comment by nawed2611** [comment]
Hey, is this still open to work on?

**Comment by dsherret** [comment]
@nawed2611 no, this work has already been started.

**Comment by nnmrts** [comment]
Can we track this work somewhere? Is there a PR for this other than https://github.com/denoland/deno_media_type/pull/4?
